### ISSUE TITLE
64 Bit Support

### ISSUE THREAD

**Issue by 26chairs** [lolp1/Process.NET#11]
I really hate to log an issue to do this (sorry), but I can't seem to find an email address to contact you anywhere. I'm trying to inject a 64 bit module into a 64 bit process and it's not working. Is 64 bit support limited to the Process.NET64 Nuget package? The 64 bit package doesn't seem to have the latest changes, so before assuming that I need to use it if I need 64 bit support, I figured I'd ask. I'm thinking it's entirely possible I'm just doing something dumb.

**Comment by lolp1** [comment]
@26chairs Nope, you are right there is an issue with injecting DLLS currently. The project has some lack in some areas for x64, currently detours and injecting are the main ones. Busy but will try to update the GitHub to latest version ASAP

**Comment by lolp1** [comment]
Working on fixing all that in the next week or two.

**Comment by 26chairs** [comment]
Neat. I really like the way the library is written and was able to learn a
bunch by looking at the code; enough to do what I was trying to do anyways.
Once injection works for x64 processes, I'll definitely have to look into
replacing the stuff I threw together for the project I'm working on with
your library instead.

**Comment by johmarjac** [comment]
Does AssemblyFactory.Execute() has issues on x64 too? I cannot seem to get it working, I get "Could not create Thread at 0x400000" Win32 Exception.

**Comment by dga711** [comment]
Is this still a thing? Any library recommended for detours in x64?

**Comment by Hi-ImKyle** [comment]
Poking this issue because I would like this issue solved. _sorry for the poke.._

**Comment by Ashlanfox** [comment]
any news on the issue ?

**Comment by adamk33n3r** [comment]
I was trying to use the 64 bit version but `ProcessSharp.Memory` is null so when I try to create a detour i get a null reference exception

**Comment by laurentksh** [comment]
@adamk33n3r  Sorry to say this but the author of the library seems to have abandoned this project which is quite sad since it's really useful..

**Comment by adamk33n3r** [comment]
@laurentksh ya it seems so :( do you know of any other good libraries that do detours? I know that microsoft has their own detours thing but it looked really clunky so I was loving the simplicity of this one. Any similar?

**Comment by laurentksh** [comment]
Nope, I too tried to do detours a few months ago but I ended up giving up because of an issue with the library.

**Comment by lolp1** [comment]
@adamk33n3r @laurentksh 

I just added support for x64/x86 detours for you guys :)

See Detour32/Detour64 classes below:
https://github.com/lolp1/Process.NET/tree/master/src/Process.NET/Applied/Detours

**Comment by laurentksh** [comment]
wow that's very nice of you, thanks !
It seems you don't get notified unless we mention you is that right ? I had the same issue with the owner of another repo, everyone complained in an Issue that the project wasn't working anymore until I mentioned him and then he replied.. @lolp1

**Comment by lolp1** [comment]
> wow that's very nice of you, thanks !
> It seems you don't get notified unless we mention you is that right ? I had the same issue with the owner of another repo, everyone complained in an Issue that the project wasn't working anymore until I mentioned him and then he replied.. @lolp1

I actually have seen the issues at some point, the problem is time and motivation. I just so happen to see this one right around when I was cleaning up some old code and had the updated version handy, so figured I'd include it. 

I still want to add-in support for x64 DLL injection and remote code execution, as those are the two primary things lacking x64 support and are fairly straight forward to fix. The DLL injection issue for example is just due to using non IntPtr types in a couple places I think.

**Comment by FlyCrafting** [comment]
Don't you plan to update the package on Nuget?

**Comment by laurentksh** [comment]
Yeah and it would also be nice to have the project use .net standard 2.0, so that projects with .NET Core can use it !

**Comment by jjbrunton** [comment]
Any chance of getting x64 mnemonics support?
